sdk/cosmos/azure_data_cosmos/src/models/mod.rs

```diff
@@ -3,37 +3,74 @@
 
 //! Model types sent to and received from the Cosmos DB API.
 
-use azure_core::{
-    date::{ComponentRange, OffsetDateTime},
-    Continuable, Model,
-};
-use serde::{Deserialize, Serialize};
+use azure_core::{date::OffsetDateTime, Continuable, Model};
+use serde::{de::DeserializeOwned, Deserialize, Deserializer};
 
 #[cfg(doc)]
 use crate::{
     clients::{ContainerClient, ContainerClientMethods, DatabaseClientMethods},
     CosmosClientMethods,
 };
 
-/// Represents a timestamp in the format expected by Cosmos DB.
-///
-/// Cosmos DB timestamps are represented as the number of seconds since the Unix epoch.
-/// Use [`CosmosTimestamp::try_into`] implementation to convert this into an [`OffsetDateTime`].
-#[derive(Serialize, Deserialize, Debug)]
-pub struct CosmosTimestamp(i64);
-
-/// Converts a [`CosmosTimestamp`] into a [`OffsetDateTime`].
-impl TryInto<OffsetDateTime> for CosmosTimestamp {
-    type Error = ComponentRange;
-
-    /// Attempts to convert this [`CosmosTimestamp`] into a [`OffsetDateTime`].
-    fn try_into(self) -> Result<OffsetDateTime, Self::Error> {
-        OffsetDateTime::from_unix_timestamp(self.0)
+/// Wrapper type used to hold items stored in a Cosmos DB Container.
+#[non_exhaustive]
+#[derive(Deserialize, Debug, Clone, Default)]
+#[serde(transparent)]
+pub struct Item<T>(
+    // This wrapper type is necessary because Response<T> needs T to implement azure_core::Model.
+    // However, we don't want to require user types to have to implement azure_core::Model.
+    // So this type does that, and handles deserializing the payload from JSON.
+    T,
+);
+
+impl<T> Item<T> {
+    /// Unwraps the inner `T` representing the item itself.
+    pub fn into_inner(self) -> T {
+        self.0
+    }
+}
+
+// The derive macro for Model doesn't currently "auto-detect" trait bounds for generic types.
+// So we have to manually implement Model
+// See https://github.com/Azure/azure-sdk-for-rust/issues/1803
+impl<T: DeserializeOwned> Model for Item<T> {
+    #[cfg(not(target_arch = "wasm32"))]
+    fn from_response_body(
+        body: azure_core::ResponseBody,
+    ) -> impl std::future::Future<Output = typespec_client_core::Result<Self>> + Send + Sync {
+        body.json()
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    fn from_response_body(
+        body: azure_core::ResponseBody,
+    ) -> impl std::future::Future<Output = typespec_client_core::Result<Self>> {
+        body.json()
+    }
+}
```

Comment:
We can actually use the derive macro, I'll remove this and use that instead. I had incorrectly assumed that the derive macro would require that `T` implement `Model` too, but unlike `Deserialize`, we don't need to be recursive in handling `Model`.

Reply:
Ah never mind, I forgot that this is necessary because of #1803